portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,32 +33,37 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  public ArrayList<String> comment_list = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String json = convertToJsonUsingGson(comment_list);
+    ArrayList<String> messages = new ArrayList<String>();
+    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      messages.add(entity.getProperty("content").toString());
+    }
+    String json_comments = convertToJsonUsingGson(messages);
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(json_comments);
   }
 
-  public String convertToJsonUsingGson(ArrayList<String> comment_arraylist) {
+  public String convertToJsonUsingGson(ArrayList<String> m) {
```

Comment:
Oh no! this got reverted back!

Reply:
It did revert back, but in my next pull request, it has changed.